src/pharmpy/modeling/summarize_individuals.py

```diff
@@ -0,0 +1,139 @@
+    ----------
+    models : list[Model]
+        Input models
+
+    Return
+    ------
+    pd.DataFrame | None
+        The summary as a dataframe
+
+    Examples
+    --------
+    >>> from pharmpy.modeling import *
+    >>> model = load_example_model("pheno")
+    >>> fit(model)
+    >>> results = run_tool(
+    >>>     'modelsearch',
+    >>>     model=model,
+    >>>     mfl='ABSORPTION(ZO);PERIPHERALS([1, 2])',
+    >>>     algorithm='reduced_stepwise'
+    >>> )
+    >>> summarize_individuals(results) # doctest: +SKIP
+
+    """
+
+    modelsDict = {model.name: model for model in models}
+
+    df = pd.concat(
+        map(
+            lambda model: groupedByIDAddColumnsOneModel(modelsDict, model),
+            models,
+        ),
+        keys=[model.name for model in models],
+        names=['model'],
+        axis=0,
+        verify_integrity=True,
+    )
+
+    return df
+
+
+def summarize_individuals_old(results: ModelSearchResults) -> Union[pd.DataFrame, None]:
+    if results.start_model is None:
+        return None
+
+    models = list(get_all_models(results))
+
+    return summarize_individuals(models)
+
+
+def get_all_models(results: ModelSearchResults) -> Iterable[Model]:
+    if results.start_model is not None:
+        yield results.start_model
+    if results.models is not None:
+        yield from results.models
+
+
+def parent_model_name(model: Model) -> str:
+    return model.parent_model
+
+
+def model_name(model: Model) -> str:
+    return model.name
+
+
+def outliers_fda_func(df: pd.DataFrame) -> float:
+    # NOTE this returns a float because we will potentially concat this
+    # with NaNs
+    return float((abs(df) > 5).sum())
+
+
+def outliers_fda(model: Model) -> Union[pd.Series, float]:
+    residuals = model.modelfit_results.residuals
+    if residuals is None:
+        return np.nan
+    else:
+        groupedByID = residuals.groupby('ID')
+        return groupedByID['CWRES'].agg(outliers_fda_func)
+
+
+def _predicted(
+    predict: Callable[[Model], pd.DataFrame], model: Model, column: str
+) -> Union[pd.Series, float]:
+    try:
+        predicted = predict(model)
+    except TypeError:
+        # TODO remove this check as a TypeError should not be thrown by a
+        # predictor, but None should be returned instead
+        return np.nan
+    if predicted is None:
+        return np.nan
+    return predicted[column]
+
+
+def predicted_residual(model: Model) -> Union[pd.Series, float]:
+    return _predicted(predict_outliers, model, 'residual')
+
+
+def predicted_dofv(model: Model) -> Union[pd.Series, float]:
+    return _predicted(predict_influential_individuals, model, 'dofv')
+
+
+def ofv(model: Model) -> pd.Series:
+    return model.modelfit_results.individual_ofv
+
+
+def dofv(modelA: Model, modelB: Model) -> pd.Series:
+    return ofv(modelA) - ofv(modelB)
+
+
+def groupedByIDAddColumnsOneModel(modelsDict: dict[str, Model], model: Model) -> pd.DataFrame:
+    df = pd.DataFrame.from_dict(
```

Comment:
I think we need to explicitly set the index here in case none of the columns are series.

Reply:
I don't know if I understand this. A column in a dataframe is always a series.

Reply:
If all the values in the dictionary are constants the 'ID' column will not exist. Currently, I do not think it is possible it would happen.